cobaya/samplers/mcmc/mcmc.py

```diff
@@ -56,11 +56,11 @@ class MCMC(CovmatSampler):
 
     # instance variables from yaml
     burn_in: NumberWithUnits
-    learn_every: NumberWithUnits
-    output_every: NumberWithUnits
+    learn_every: Union[NumberWithUnits, str]
```

Comment:
Not sure why only some of these changed. In terms of usage, optional str is probably not very helpful since should be converted at read in, maybe NumberWithUnits and more flexible check

Reply:
I changed the only ones causing errors, but, following your suggestion, I moved the problem on the type-checking side as we know the `NumberWithUnits` can come in as a string.